# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/notifications/instance-checks.js
# Commit: ed71fdc2d78d
# Full Hash: ed71fdc2d78dec085a3d153220bba4aeda89992b
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-05 04:28:59
# Regressor Bug: 1951379
# File Overlap Count: 1
# Description:
#   Bug 1951379 - Part 3: Add NotificationStorageEntry r=asuth
#   
#   This skips handling `vibrate` for now, as we don't ship it and it also complicates test setup.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240213
# ==============================================================================

diff -r 61ab7888abc4 -r ed71fdc2d78d testing/web-platform/tests/notifications/instance-checks.js
--- a/testing/web-platform/tests/notifications/instance-checks.js	Tue Mar 04 18:00:17 2025 +0000
+++ b/testing/web-platform/tests/notifications/instance-checks.js	Tue Mar 04 18:00:18 2025 +0000
@@ -6,6 +6,8 @@
       body: "This is a radio check.",
       tag: "radio_check999",
       icon: `${location.origin}/icon.png`,
+      requireInteraction: true,
+      silent: true,
       data: fakeCustomData,
   }
 ];
@@ -17,24 +19,30 @@
     n = await createFn(t);
   }, `${testTitle}: Setup`);
   promise_test(async () => {
-    assert_equals("Radio check", n.title)
+    assert_equals(n.title, "Radio check")
   }, `${testTitle}: Attribute exists with expected value: title`)
   promise_test(async () => {
-    assert_equals("ltr", n.dir)
+    assert_equals(n.dir, "ltr")
   }, `${testTitle}: Attribute exists with expected value: dir`)
   promise_test(async () => {
-    assert_equals("aa", n.lang)
+    assert_equals(n.lang, "aa")
   }, `${testTitle}: Attribute exists with expected value: lang`)
   promise_test(async () => {
-    assert_equals("This is a radio check.", n.body)
+    assert_equals(n.body, "This is a radio check.")
   }, `${testTitle}: Attribute exists with expected value: body`)
   promise_test(async () => {
-    assert_equals("radio_check999", n.tag)
+    assert_equals(n.tag, "radio_check999")
   }, `${testTitle}: Attribute exists with expected value: tag`)
   promise_test(async () => {
-    assert_equals(`${location.origin}/icon.png`, n.icon)
+    assert_equals(n.icon, `${location.origin}/icon.png`)
   }, `${testTitle}: Attribute exists with expected value: icon`)
   promise_test(async () => {
+    assert_true(n.requireInteraction);
+  }, `${testTitle}: Attribute exists with expected value: requireInteraction`)
+  promise_test(async () => {
+    assert_true(n.silent);
+  }, `${testTitle}: Attribute exists with expected value: silent`)
+  promise_test(async () => {
     assert_custom_data(n.data);
   }, `${testTitle}: Attribute exists with expected value: data`)
 }
